Add tests for LateralProfile

MarekOchocki opened the pull request.

MarekOchocki committed:
Implement methods in lateral class (#9)

* Implement methods in lateral class

- penumbra (left and right)
- field_ratio
- symmetry
- flatness_50
- flatness_90
- asymmetry
- normalize
- absolute_y

* Add comments explaining methods

- docstring explaining penumbra
- docstring explaining absolute_y
- imports sorted

* Replace constant integers with floats

* Merge absolute_y with normalize

You have to call absolute_y before calling normalize anyway.
If you call these in opposite way you'd have to call normalize again.

* Cosmetic changes to fit PEP8 style

* Cosmetic changes part 2

MarekOchocki committed:
Add tests for LateralProfile

MarekOchocki committed:
Fix tests for LateralProfile

grzanka commented on tests/lateral.py:
```diff
@@ -5,13 +5,122 @@
+        with self.assertRaises(ValueError):
+            self.data1.symmetry(-3)
+
+        self.assertAlmostEqual(self.data2.symmetry(1.0), 37.5)
```
why using `assertAlmostEqual` if the value seems to be quite exact (37.5) ?

MarekOchocki committed:
Change assertAlmostEqual to assertEqual in some places

MarekOchocki committed:
Add methods in DepthProfile (#13)

* Add methods in DepthProfile

- range
- FWHM
- max_plat_ratio
- distal_falloff
- modulation
- normalize

* Add comment and rename some methods

- remove range, because function width() from Profile does the same
- explain modulation
- rename fwhm

* Add ran method and remove fwhm

fwhm is already implemented in Profile

MarekOchocki committed:
Lateral profile (#12)

* Implement methods in lateral class

- penumbra (left and right)
- field_ratio
- symmetry
- flatness_50
- flatness_90
- asymmetry
- normalize
- absolute_y

* Add comments explaining methods

- docstring explaining penumbra
- docstring explaining absolute_y
- imports sorted

* Replace constant integers with floats

* Merge absolute_y with normalize

You have to call absolute_y before calling normalize anyway.
If you call these in opposite way you'd have to call normalize again.

* Cosmetic changes to fit PEP8 style

* Cosmetic changes part 2

* Adding allow_cast in normalize

- allow_cast in normalize
- fixed penumbras

* Cosmetic changes to fit PEP8

* Add handling with corrupted data

* Fix raising exception in normalize()

* Change implementations to be more accurate

- Fix ignoring part of the profile in the middle in asymmetry()
- Add searching for max/min value on borders in flatness_90 and flatness_50

* Extend comments

Additionally, change checking for NaN from d != d to np.isnan(d) to make it less confusing

* Change code to fit PEP8

grzanka commented:
@MarekOchocki please:
 - checkout master
 - pull
 - rebase feature/11-lateral-tests to master
 - push --force

or use any other method to include in this branch content from merged PR #12 and #13

grzanka commented:
Not valid
